src/fastoad/cmd/api.py

```diff
@@ -328,7 +329,7 @@ def generate_inputs(
 
 def list_variables(
     configuration_file_path: str,
-    out: Union[IO, str] = None,
+    out: Union[TextIOBase, str] = None,
```

Comment:
The reason you change from IO to TextIOBase is because the IO type was too generic since we only expect text files ?

Reply:
The reason is that I noticed that given `f = open(...)`, `isinstance(f, IO)` returns `False` whereas `isinstance(f, IOBase)` returns `True` (and here `TextIOBase` to, indeed, enforce text files).
So it looked like a good idea to change that.

You noticed an inconsistency with other parts of the code, where I kept `IO`. The reason is that I realized later that PyCharm type checker was complaining unless `IO` was in the type hint.

I did some tests with mypy: it never complains when `IO` is in the type hint, but can complain for the type `IOBase` if provided `sys.stdout`

I think now I will get back to having `IO` in type hints. Looks like the expected way. We just have to keep in mind that `isinstance` checks must not use `IO` (which is rather odd, compared to other type hints, but I/O flows have always been a messy subject in Python, I guess 😅).